### ISSUE TITLE
error on LabelBinarizer when feature has two labels 

### ISSUE THREAD

**Issue by mazurkin** [jpmml/jpmml-sklearn#36]
Error appears when sklearn.preprocessing.LabelBinarizer is used and feature has two labels. In that case LabelBinarizer switches to "binary mode" and produces only one column.

So later we have such an error in DataFrame. updatePMML()

```
if (inputNames.size() != numberOfInputs) {
   throw new IllegalArgumentException();
}
```

The problem is discovered on pretty old (1.0-SNAPSHOT) version of jpmml-sklearn but I see that sklearn.preprocessing.LabelBinarizer is the same. So I suppose the problem still exists. Going to check with the latest version of jpmml-sklearn later

**Comment by MatiasSanchezCabrera** [comment]
You should use sklearn.preprocessing.LabelEncoder for binary variables.

**Comment by mazurkin** [comment]
Ok, I'll take a look on LabelEncoded. Also I'll to explain in details the problem.

I have a feature with name 'imprtype' that could have values like 'B', 'V', 'X', 'N' and I'd like to expand this single feature with 4 columns that have 0/1 values with LabelBinarizer. 

Everything works well until my training set has only two labels: 'V', 'B'. In this case LabelBinarizer switches to "binary" mode and produces only one column instead of expected two columns. So later I have a problem with exporting this model with jpmml.

So my feature is not binary. It could have many labels. But if (for some reason) only two labels are registered then I got an error.

**Comment by MatiasSanchezCabrera** [comment]
The thing is that you can't say a column is polynomial if it is binary in your dataset. You might need to check this before you train your pipeline before exporting the model to pmml.

**Comment by mazurkin** [comment]
Direct replacing LabelBinarizer with LabelEncoder produces warning and then error. Need to research the problem more deeply. 

'''
/home/datascience/.conda/envs/models/lib/python2.7/site-packages/sklearn/preprocessing/label.py:129: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
'''

From other point of view - why not use LabelBinarizer with two labels? It works in Python perfectly, the problem raises only on model export. 

I made my own TrueLabelBinarizer that produces two columns instead of single column for two labels. But unfortunately jpmml knows nothing about custom transformer classes. 

http://stackoverflow.com/questions/31947140/sklearn-labelbinarizer-returns-vector-when-there-are-2-classes

**Comment by paubelsan** [comment]
I agree with @mazurkin, I have exactly the same problem!
